src/WebJobs.Extensions.DocumentDB/Config/IDocumentDBServiceFactory.cs

```diff
@@ -5,6 +5,6 @@ namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
 {
     internal interface IDocumentDBServiceFactory
     {
-        IDocumentDBService CreateService(string connectionString);
+        IDocumentDBService CreateService(string connectionString, string desiredConnectionMode);
```

Comment:
Nit: can you make this take the enum? That way services don't have to do the conversion themselves.

Reply:
Sure thing!